kedro/io/data_catalog.py

```diff
@@ -311,6 +358,28 @@ def _get_dataset(
 
         return data_set
 
+    def _get_resolved_dataset(
+        self, dataset_name: str, matched_pattern: str
+    ) -> AbstractDataSet:
+        result = parse(matched_pattern, dataset_name)
```

Comment:
Not necessarily needed to do this twice, I could also save the `result` returned by parse earlier in the cache along with the name  in the `match_name_against_pattern()` method but I didn't want to use tuples in the cache dict. Doing this again here isn't really a performance bottleneck since this only happens once.